### PR TITLE
[iOS] Fixes crash in CollectionViews when using DataTemplateSelector as a DataTemplate when used in GroupHeaders and or GroupFooters

### LINKED ISSUE #8647: [Bug] Crash on iOS when using DataTemplateSelector as a GroupingHeaderTemplate in a CollectionView
### Crash on iOS when using DataTemplateSelector as a GroupingHeaderTemplate or GroupingFooterTemplate

This issue was first reported by @timahrentlov in Issue #8609
I just provide the Minimal repo to reproduce the bug here, and a suggestion on where I think the bug ln the Xamarin Code is.

### Steps to Reproduce

1. Use a DataTemplateSelector as a GroupHeaderTemplate in a CollectionView
2. Enable IsGrouped and fill data

### Expected Behavior
Display Groupheaders using the DataTemplate provided by the DataTemplateSelector

### Actual Behavior
Android: As expected.
iOS: Crash with "System.InvalidOperationException: LoadTemplate should not be null"

- Version with issue: v4.3.0.991211
- Last known good version: Unknown
- IDE: VS 2019 16.3.10
- Platform Target Frameworks:
  - iOS:  8
  - Android: 9.0

### Reproduction Link

https://github.com/MariovanZeist/Xamarin.CollectionViewGrouping

Or use zip file:

[Xamarin.CollectionViewGrouping.zip](https://github.com/xamarin/Xamarin.Forms/files/3886383/Xamarin.CollectionViewGrouping.zip)

The problem is in the Xamarin.Forms.Platform.iOS.GroupableItemsViewController class
specifically in the UpdateTemplatedSupplementaryView function.

This function gets the group header or footer DataTemplate, and then calls a cell.Bind override that does not check if the DataTemplate is actually a DataTemplateSelector (code is around line 105)
hence the crash.

### PR DESCRIPTION
**Pull request by MariovanZeist**
### Description of Change ###

CollectionViews with GroupHeaders and or GroupFooters will be able to use DataTemplateSelectors and DataTemplates. (Used to crash when you would set a Header/ Footer to a DataTemplateSelector)

CollectionViews now will only display A Group Header or Footer when IsGrouped is set to true AND a datatemplate is set for that Header and or Footer. (It used to display Header and Footer regardless of the fact that a DataTemplate was selected)

Some minor code cleanup.

### Issues Resolved ### 
- fixes #8647 
### API Changes ###
None

Added:
 None

### Platforms Affected ### 
- iOS

### Behavioral/Visual Changes ###
An iOS user will be able to use A DataTemplateSelector in a collectionview without crashing the App/

### Testing Procedure ###
Open ControlGalary App, select test 8647 and see that it doesn't crash, and only shows Group Headers (Not footers)

Video: (QuickTime)

[8647-TestRun.zip](https://github.com/xamarin/Xamarin.Forms/files/3892494/8647-TestRun.zip)

### PR Checklist ###

- [x] Targets the correct branch
- [x] Tests are passing (or failures are unrelated)

### CODE REVIEW

File: Xamarin.Forms.Platform.iOS/CollectionView/TemplatedCell.cs [hunk 1]
```diff
@@ -64,21 +64,16 @@ public override void ConstrainTo(nfloat constant)
...
-
-			Bind(template, bindingContext, itemsView);
-		}
-
 		public void Bind(DataTemplate template, object bindingContext, ItemsView itemsView)
 		{
 			var oldElement = VisualElementRenderer?.Element;
 
-			if (template != _currentTemplate)
+			var itemTemplate = template;
```
Comment (VladislavAntonyuk) [inline Xamarin.Forms.Platform.iOS/CollectionView/TemplatedCell.cs]: var itemTemplate = template.SelectDataTemplate(bindingContext, itemsView);

Comment (MariovanZeist) [reply Xamarin.Forms.Platform.iOS/CollectionView/TemplatedCell.cs]: Yeah a lot cleaner

**Comment by jfversluis** [comment]
Awesome stuff @MariovanZeist !

I think the code looks good, just two things:
- could you give the PR here a more meaningful title than just the issue number? That title will go into the commit history.
- Could you rebase this onto the 4.4 branch? Since it's a bug we would like to get it resolved in our current stable branch

Thanks!

**Comment by MariovanZeist** [comment]
@jsuarezruiz, @StephaneDelcroix  
I fixed a merge conflict. that I got after retargeting my PR. As this was my first time using github merge (And first time contributing at all), I hope I didn't do something wrong.

**Comment by jfversluis** [comment]
It looks like you got a lot of other commits and changes in there now. You want to do a rebase, are you familiar with those?

**Comment by MariovanZeist** [comment]
@jfversluis I thought I rebased it, although this was my first try doing it in GitHub directly, as opposed to doing it in VS. I guess I did something wrong.  Can you give me some pointers?

**Comment by jfversluis** [comment]
I don't think you can do it through GitHub.

You want to go to your fork and branch and probably open up a terminal unless you have some other tool that can do it. Then run `git rebase -i origin/4.4.0` the `-i` is optional, it will show you an overview of all the commits that it is going to replay on top of the target branch, in this case `origin/4.4.0`.

After it's done and you resolved the conflicts you force push the new branch which is basically the current 4.4.0 branch with your commits on top.

You can see me doing it [here](https://www.youtube.com/watch?v=Nt-BYL1WhC0&feature=youtu.be&t=3627)

**Comment by MariovanZeist** [comment]
Alright, I had to do some Git Magic as I seemed to have included commits made to master into the 4.4.0 branch. But it looks clean again now.

**Comment by jfversluis** [comment]
Yes! Looks all good now! Thanks! 😄

**Comment by jsuarezruiz** [comment]
Looks nice but could you review the conflict? Thanks.

**Comment by MariovanZeist** [comment]
@jsuarezruiz I am not seeing a conflict (Although I might not be looking in the correct place) or has it been resolved already?

**Comment by jfversluis** [comment]
@MariovanZeist I think it was solved by one of us for you and now your change is merged 🎉 

Congrats and thank you for your contribution!

**Pull request merged**